src/hooks/useNetworkSwitch.ts

```diff
@@ -1,38 +1,54 @@
+import { Connector } from '@web3-react/types'
 import { useCallback } from 'react'
-
-import { CustomNetworkConnector } from '../connectors/CustomNetworkConnector'
-import { CustomWalletLinkConnector } from '../connectors/CustomWalletLinkConnector'
-import { NETWORK_DETAIL } from '../constants'
-import { switchOrAddNetwork } from '../utils'
-
-import { useActiveWeb3React, useUnsupportedChainIdError } from '.'
-
-export type UseNetworkSwitchProps = {
-  onSelectNetworkCallback?: () => void
+import { useDispatch } from 'react-redux'
+
+import { networkConnection, walletConnectConnection } from '../connectors'
+import { getConnection, isChainSupportedByConnector } from '../connectors/utils'
+import { AppDispatch } from '../state'
+import { setConnectorError } from '../state/application/actions'
+import { getErrorMessage } from '../utils/getErrorMessage'
+import { getNetworkInfo } from '../utils/networksList'
+import { useWeb3ReactCore } from './useWeb3ReactCore'
+
+const switchNetwork = async (connector: Connector, chainId: ChainId) => {
+  if (!isChainSupportedByConnector(connector, chainId)) {
+    throw new Error(`Chain ${chainId} not supported for connector (${typeof connector})`)
+  } else if (connector === walletConnectConnection.connector || connector === networkConnection.connector) {
+    await connector.activate(chainId)
+  } else {
+    const info = getNetworkInfo(chainId)
+    const addChainParameter = {
+      chainId,
+      chainName: info.chainName,
+      rpcUrls: info.rpcUrl,
+      nativeCurrency: info.nativeCurrency,
+      blockExplorerUrls: info.blockExplorerUrls,
+    }
+    await connector.activate(addChainParameter)
+  }
 }
 
-export const useNetworkSwitch = ({ onSelectNetworkCallback }: UseNetworkSwitchProps = {}) => {
-  const { connector, chainId, account } = useActiveWeb3React()
-  const unsupportedChainIdError = useUnsupportedChainIdError()
+export const useNetworkSwitch = () => {
+  const { connector } = useWeb3ReactCore()
+  const dispatch = useDispatch<AppDispatch>()
 
   const selectNetwork = useCallback(
-    (optionChainId?: ChainId) => {
-      if (optionChainId === undefined || (optionChainId === chainId && !unsupportedChainIdError)) return
-      if (!account && !unsupportedChainIdError && connector instanceof CustomNetworkConnector)
-        connector.changeChainId(optionChainId)
-      else if (!account && unsupportedChainIdError && connector instanceof CustomNetworkConnector)
-        connector.switchUnsupportedNetwork(NETWORK_DETAIL[optionChainId])
-      else if (connector instanceof InjectedConnector)
-        switchOrAddNetwork(NETWORK_DETAIL[optionChainId], account || undefined)
-      else if (connector instanceof CustomWalletLinkConnector)
-        connector.changeChainId(NETWORK_DETAIL[optionChainId], account || undefined)
-
-      if (onSelectNetworkCallback) onSelectNetworkCallback()
+    async (chainId: ChainId) => {
+      if (!connector) return
```

Comment:
`connector` is `Connector` type so you can remove this check

Reply:
fixed [08e6d75](https://github.com/levelkdev/swapr-dapp/pull/1330/commits/08e6d75ca9c55e8ad7740066bd2f64f410db9053)